packages/kit/src/lib/HookSystem.ts

```diff
@@ -176,24 +241,40 @@ export class HookSystem<
 		return hooks;
 	}
 
+	/**
+	 * Returns list of hooks for a given name
+	 */
+	hooksForName(name: string): RegisteredHook[] {
+		return this._registeredHooks[name] ?? [];
+	}
+
 	createScope<TExtraArgs extends unknown[] = never[]>(
 		owner: string,
 		extraArgs: [...TExtraArgs],
-		meta: Partial<RegisteredHookMeta> = {},
-	): CreateScopeReturnType<THookFns, TExtraArgs> {
+		scopeMeta: Partial<RegisteredHookMeta> = {},
+	): CreateScopeReturnType<THooks, TExtraArgs, THookFns, THookMetas> {
 		return {
 			hook: <TName extends Extract<keyof THookFns, string>>(
 				name: TName,
 				hookFn: WithExtraArgs<THookFns[TName], TExtraArgs>,
+				...args: HookMetaArg<THookMetas[TName]>
```

Comment:
Clever way to handle conditional required arguments. 😉

An overloaded type is probably the more correct way to handle this, but I found `siroc` (or some underlying plugin) doesn't work with overloads. This is a good secondary solution.